BEIMA.Client/src/services/AddDevice.js

```diff
@@ -7,9 +7,21 @@ const API_URL = process.env.REACT_APP_API_URL;
  * @param The device details to add to the DB
  * @return Error message or the inserted device ID
  */
-export default async function addDevice(deviceDetails) {
+export default async function addDevice(deviceDetails, photo, files) {
+  // setup the multiform request data
   let formData = new FormData();
   formData.append("data", JSON.stringify(deviceDetails));
+  formData.append("photo", photo);
+  const tempFiles = [...files]
+
+  // add each additional file to the files key
+  if(files){
+    tempFiles.map((file, index) => {
+      formData.append(`files${index}`, file);
```

Comment:
Just trying to understand here, so each file is going in their own bucket? I thought we were putting all the extra files in the 'files' bucket? Also, does this include the file names when it sends these over?

Reply:
Yes I believe that is still correct, the files going into one bucket. I started off using the code Kenny suggested in the chat, which had it like "files[${index}]", but it looked like the backend read each file with the name in the formData as "files[0]" etc. I can try removing the index variable and just appending to the files, I don't think append overrides anything in there

- [ ] try without index

Reply:
Appending to files works